app/models/main.py

```diff
@@ -1371,6 +1371,28 @@ class query_class(BaseQuery):
         def has_statuses(self, *statuses):
             return self.filter(Brief.status.in_(statuses))
 
+        def data_key_contains_value(self, k, v):
+            return self.filter(Brief.data[k].astext.contains(u'"{}"'.format(v)))  # Postgres 9.3: use string matching
+
+        def has_inclusive_date_range(self, attr, start_date=None, end_date=None):
```

Comment:
As a model method this actually does 3 things. I'd rather see it as 3 methods on the model. before, after and between. That way they can be tested separately and processing the args we get from the get request could be done in the view. Ie (if not (start_date or end_date), if not end_date, if not start_date) then call the respective model methods in the view